List_Representation/runs/config1.txt
Experiment ID                :  Traveling_Salesman
Problem Type                 :  TSP
Data Input File Name         :  att48.tsp
Number of Runs               :  5
Generations per Run          :  200
Population Size              :  2000
Selection Method             :  2
Fitness Scaling Type         :  4
Min or Max Fitness           :  min
Crossover Type               :  1
Crossover Rate               :  0.8
Mutation Type                :  1
Mutation Rate                :  0.4
Random Number Seed           :  75982
Number of Genes/Points       :  48
Size of Genes                :  9


 R   1 G   0 118460 157684.031   10799.08
 R   1 G   1 111292 152168.604  10568.495
 R   1 G   2 111863 147775.785   10486.81
 R   1 G   3  99283 144502.777  10438.494
 R   1 G   4 104597   141193.1  10460.158
 R   1 G   5 101121 138409.872  10061.319
 R   1 G   6 102334 136027.625  10174.513
 R   1 G   7 104953 134100.781  10027.993
 R   1 G   8  98469 132626.271  10165.753
 R   1 G   9  98328 130839.469  10234.243
 R   1 G  10  98328 128887.507  10100.934
 R   1 G  11  98460 127603.366  10327.772
 R   1 G  12  97010 126745.085  10315.357
 R   1 G  13  96753 125755.398  10104.171
 R   1 G  14  96283 124563.814  10359.695
 R   1 G  15  93948 123323.467  10243.429
 R   1 G  16  93013 122042.393  10158.761
 R   1 G  17  91132 121342.308  10002.394
 R   1 G  18  86703 120465.525  10207.105
 R   1 G  19  87509 119886.493  10518.522
 R   1 G  20  87047 118842.336  10232.446
 R   1 G  21  87047 118073.283  10217.133
 R   1 G  22  86766 117809.169  10314.632
 R   1 G  23  88650 116604.358   9968.946
 R   1 G  24  88650 116173.043  10211.775
 R   1 G  25  85042 115316.439   9949.369
 R   1 G  26  86177 114696.033   9878.628
 R   1 G  27  86834 114450.284   9653.634
 R   1 G  28  86955 114213.605   9957.902
 R   1 G  29  83419 114158.079  10000.429
 R   1 G  30  87370 113787.898  10120.353
 R   1 G  31  85948 114106.363  10302.512
 R   1 G  32  84826 114132.129  10536.181
 R   1 G  33  81407 113160.809  10609.977
 R   1 G  34  81032 112356.694  10699.462
 R   1 G  35  84204 111642.393  10601.242
 R   1 G  36  83307 110834.543  10316.426
 R   1 G  37  85191 110378.189  10372.735
 R   1 G  38  77310 109810.427  10260.348
 R   1 G  39  77568 109250.052  10103.225
 R   1 G  40  81721 108919.641   10221.82
 R   1 G  41  80542 108727.709  10003.714
 R   1 G  42  81305 108335.716  10004.685
 R   1 G  43  77252 108047.586  10052.814
 R   1 G  44  77252 107442.722  10033.837
 R   1 G  45  74096 106923.504   9994.686
 R   1 G  46  74529 106381.399  10145.117
 R   1 G  47  77675 105642.892  10216.993
 R   1 G  48  79971 105537.629  10024.466
 R   1 G  49  77419 105374.231   9762.051
 R   1 G  50  77888 105114.771   9696.819
 R   1 G  51  77888 104575.495   9829.538
 R   1 G  52  76621 104317.279  10070.715
 R   1 G  53  76258 103924.762   9588.082
 R   1 G  54  76258 103894.446   9780.989
 R   1 G  55  79359 103454.369   9890.596
 R   1 G  56  74815 102707.082   9672.588
 R   1 G  57  74815 102724.055   9928.822
 R   1 G  58  78503 102295.808  10082.762
 R   1 G  59  75864 101477.185   9755.832
 R   1 G  60  74787 101033.186   9726.921
 R   1 G  61  73800 101144.168  10074.891
 R   1 G  62  73800 100698.116   9519.384
 R   1 G  63  75690 100555.218   9582.988
 R   1 G  64  75278  100145.88   9487.215
 R   1 G  65  75278 100302.847   9826.672
 R   1 G  66  71716  99823.406   9574.532
 R   1 G  67  73424   99286.72   9419.386
 R   1 G  68  70686  99100.332   9426.771
 R   1 G  69  73505  99347.213   9765.802
 R   1 G  70  73069   99534.98   9719.121
 R   1 G  71  69548  99316.345   9692.531
 R   1 G  72  75574  99051.333   9961.469
 R   1 G  73  73476  98648.447   9900.919
 R   1 G  74  71569   98677.87   9647.033
 R   1 G  75  70334  98261.407   9542.659
 R   1 G  76  70446  97820.954   9582.785
 R   1 G  77  70446  97567.585   9504.131
 R   1 G  78  73493  97408.587   9760.281
 R   1 G  79  73044  97014.865    9687.07
 R   1 G  80  72361   96713.58   9571.596
 R   1 G  81  73543  96630.929   9397.229
 R   1 G  82  70856  97042.197   9610.557
 R   1 G  83  69431  97428.529   9902.624
 R   1 G  84  70865  96784.516    9633.09
 R   1 G  85  66615  96172.691   9722.157
 R   1 G  86  70577  95456.875   9805.709
 R   1 G  87  70577  95323.095   9863.063
 R   1 G  88  70336  94924.905    9757.66
 R   1 G  89  70515  95031.998   9646.563
 R   1 G  90  67503  94977.139   9493.856
 R   1 G  91  69679  94877.349   9036.017
 R   1 G  92  66389  95028.204   9159.648
 R   1 G  93  68245  95462.943   9525.323
 R   1 G  94  69661  95217.486   9748.408
 R   1 G  95  69661  95073.703   9403.987
 R   1 G  96  69965  95109.602   9506.036
 R   1 G  97  70575  95127.184   9419.012
 R   1 G  98  66605  94975.197   9606.331
 R   1 G  99  69658   94984.65   9444.766
 R   1 G 100  69658   94770.49   9404.906
 R   1 G 101  71472  94996.065   9701.476
 R   1 G 102  71261   94728.12    9770.96
 R   1 G 103  70757   94451.39   9348.221
 R   1 G 104  70880  94443.289   9538.772
 R   1 G 105  70684  94120.095   9334.312
 R   1 G 106  69283  94426.241   9502.352
 R   1 G 107  67522  94597.087    9686.76
 R   1 G 108  69782  94666.151   9842.231
 R   1 G 109  68485  94243.669   9654.092
 R   1 G 110  68485  94625.346  10102.678
 R   1 G 111  66878  94121.619   9976.008
 R   1 G 112  68023  94013.003   9618.078
 R   1 G 113  70468  93550.433   9975.979
 R   1 G 114  69338  93066.082   9511.705
 R   1 G 115  68168  93024.593   9507.853
 R   1 G 116  70242  93047.746   9731.148
 R   1 G 117  65214  93045.502   9630.142
 R   1 G 118  65214  92913.714   9623.353
 R   1 G 119  68226  92942.798   9455.815
 R   1 G 120  68226  92531.492    9296.41
 R   1 G 121  68525  92843.963   9335.625
 R   1 G 122  68942  92971.056   9399.356
 R   1 G 123  68028  93234.264   9689.229
 R   1 G 124  68043  93451.807   9480.622
 R   1 G 125  69043  93750.527   9520.277
 R   1 G 126  67452  94026.219   9682.094
 R   1 G 127  67234   93718.29   9829.631
 R   1 G 128  65707  93726.814   9430.823
 R   1 G 129  68302  94121.144   9617.894
 R   1 G 130  68302  94302.764   9450.384
 R   1 G 131  68099  94484.865   9922.411
 R   1 G 132  67027  93886.884   9829.763
 R   1 G 133  65957  93657.464   9729.669
 R   1 G 134  68206  93805.305   9706.541
 R   1 G 135  67460  93703.805   9474.008
 R   1 G 136  68329  93395.339   9368.448
 R   1 G 137  70556  93805.978   9437.926
 R   1 G 138  69678  94205.765   9391.676
 R   1 G 139  70297  94689.746   9563.011
 R   1 G 140  69299  94836.916   9587.338
 R   1 G 141  70297    94730.1    9391.78
 R   1 G 142  67351  94578.161   9416.868
 R   1 G 143  70342  95187.611   9783.112
 R   1 G 144  70342  95380.569    9687.99
 R   1 G 145  70432  95522.917    9624.32
 R   1 G 146  71532   95948.74   9704.434
 R   1 G 147  68895  96366.305   9621.561
 R   1 G 148  71700  96727.012   9853.325
 R   1 G 149  71449  96616.342   9545.433
 R   1 G 150  70415  96462.345   9719.306
 R   1 G 151  65200  96583.478   9886.896
 R   1 G 152  71569  96892.828   9740.901
 R   1 G 153  71799  96644.126   9553.018
 R   1 G 154  71799   96635.62   9432.553
 R   1 G 155  71799  96726.073   9813.628
 R   1 G 156  69418  96647.906  10118.489
 R   1 G 157  73288  96201.328   9768.925
 R   1 G 158  73288  96142.159   9745.191
 R   1 G 159  70703  95689.816    9454.68
 R   1 G 160  71432  95719.406   9534.965
 R   1 G 161  71552   96287.53   9885.232
 R   1 G 162  71683  95933.107   9737.475
 R   1 G 163  73245  95838.593   9401.355
 R   1 G 164  73020  95703.732    9337.92
 R   1 G 165  70439  95625.052   9623.901
 R   1 G 166  70439  95438.363   9764.408
 R   1 G 167  70439  95093.082   9603.009
 R   1 G 168  69461  95319.648   9735.408
 R   1 G 169  66811  95297.723  10086.612
 R   1 G 170  66993  94827.501   9870.856
 R   1 G 171  68752  94993.834  10245.305
 R   1 G 172  68907  94757.147   9831.708
 R   1 G 173  71806  94933.009   9749.377
 R   1 G 174  71806  95019.934    9659.36
 R   1 G 175  72302  94912.761   9653.689
 R   1 G 176  69062  95171.912    9899.78
 R   1 G 177  64635  94810.826   9784.506
 R   1 G 178  64635  94944.262   9860.577
 R   1 G 179  69486  94943.446   9876.951
 R   1 G 180  71502  95310.511   9999.924
 R   1 G 181  66676  95214.865  10123.389
 R   1 G 182  70137  94859.254   9930.923
 R   1 G 183  70137  94672.396   9961.331
 R   1 G 184  68610  94522.897  10086.548
 R   1 G 185  70459  94638.651   9987.139
 R   1 G 186  65329  94328.892   9741.892
 R   1 G 187  65329  94095.858  10140.553
 R   1 G 188  68507  93533.056   9572.896
 R   1 G 189  70223  93598.225   9627.921
 R   1 G 190  69612  93643.263    9757.86
 R   1 G 191  65310  93438.854   9752.388
 R   1 G 192  68399  93193.627   9599.769
 R   1 G 193  68033  93229.198   9857.398
 R   1 G 194  69336   92675.02   9365.608
 R   1 G 195  67767  92369.543    9278.18
 R   1 G 196  66755  92095.157   9343.265
 R   1 G 197  66755  92017.155   9346.009
 R   1 G 198  64669  91871.481   9376.495
 R   1 G 199  64669  92012.532   9460.487
1    177         15         33         40         19          6          7         44         43         37         36         31         38          9         22         16          1         41         11         21         46         18         28         17         27         30         20         14         13          3         34          5         42         25          8         47         12         32         39         48          2         29         10          4         26         35         45         24         23   RawFitness
                64635

 R   2 G   0 115587 157965.861  11115.456
 R   2 G   1 115489 152553.268  10908.621
 R   2 G   2 109732 148257.985  10183.745
 R   2 G   3 107707 144843.427  10096.917
 R   2 G   4 109732 141911.426  10166.692
 R   2 G   5 106422 139400.552  10345.312
 R   2 G   6 106935   137169.0   9883.922
 R   2 G   7 103784 135123.378   9999.979
 R   2 G   8 105569 133472.234  10015.999
 R   2 G   9 102043 132338.892  10154.743
 R   2 G  10 102476 130523.924   9973.436
 R   2 G  11  97700 129866.886  10148.439
 R   2 G  12  93675 128593.261  10423.478
 R   2 G  13  97924 127546.154  10196.318
 R   2 G  14  96342 126171.298  10242.797
 R   2 G  15  92308  125058.36  10037.905
 R   2 G  16  93905 124421.197    9908.76
 R   2 G  17  93905 123716.867  10141.799
 R   2 G  18  95878 123181.253  10444.088
 R   2 G  19  91278 122065.662    10517.2
 R   2 G  20  87464  120784.17  10691.586
 R   2 G  21  87464 119815.673    10452.8
 R   2 G  22  89496 118916.516  10404.713
 R   2 G  23  86686 118081.226  10623.893
 R   2 G  24  89397 117459.322  10349.523
 R   2 G  25  86923 116694.606  10548.846
 R   2 G  26  87403 116011.666  10476.266
 R   2 G  27  87308 115321.692   9947.222
 R   2 G  28  89136 114737.064  10228.124
 R   2 G  29  84183 114600.077  10197.879
 R   2 G  30  84571 113955.056   10180.79
 R   2 G  31  75672 113410.118  10306.845
 R   2 G  32  75672 113056.514  10109.777
 R   2 G  33  85231 112689.278  10309.262
 R   2 G  34  77335 111951.147  10428.699
 R   2 G  35  82035 111812.366  10662.352
 R   2 G  36  79839 110738.596   10287.08
 R   2 G  37  80328 109913.628  10253.826
 R   2 G  38  79868 109106.698  10273.871
 R   2 G  39  79868 108766.042  10042.716
 R   2 G  40  80519 107883.105   9857.574
 R   2 G  41  83219 107783.556  10083.053
 R   2 G  42  82866 107375.795  10044.762
 R   2 G  43  77905 107311.322  10456.463
 R   2 G  44  76555 106596.311  10169.755
 R   2 G  45  77450 106039.845  10011.486
 R   2 G  46  78476 105411.858   9643.851
 R   2 G  47  77501 105178.069   9626.441
 R   2 G  48  78724  104648.99   9407.031
 R   2 G  49  76795 104384.996   9592.658
 R   2 G  50  73972 104230.716  10065.782
 R   2 G  51  74228  103580.75   9733.642
 R   2 G  52  78202 104114.806  10112.769
 R   2 G  53  78205 103855.059   9877.149
 R   2 G  54  78277 104275.999  10055.119
 R   2 G  55  71285 103823.996  10121.535
 R   2 G  56  73086 103402.477  10015.316
 R   2 G  57  74634 103155.995   9805.259
 R   2 G  58  74634 103219.403  10226.453
 R   2 G  59  75345 102910.838  10062.442
 R   2 G  60  77348 102783.033  10272.411
 R   2 G  61  73244 102298.093   9861.978
 R   2 G  62  73244 102059.979   9869.417
 R   2 G  63  74291 101923.239   9807.041
 R   2 G  64  78020 101556.912   9542.997
 R   2 G  65  75128 101759.042   9740.448
 R   2 G  66  75094 101746.688   9906.907
 R   2 G  67  75880 101548.687   9719.107
 R   2 G  68  71444 101564.653   9343.114
 R   2 G  69  75398 101368.642   9519.613
 R   2 G  70  75398 100971.299   9236.572
 R   2 G  71  74947 101434.123   9485.932
 R   2 G  72  75394  101113.74    9674.79
 R   2 G  73  76607 101032.008   9900.976
 R   2 G  74  74383 100608.583   9725.203
 R   2 G  75  74383 100526.137   9879.276
 R   2 G  76  74670 100341.177    9723.63
 R   2 G  77  75185  99859.607   9440.967
 R   2 G  78  75185  99442.766   9486.966
 R   2 G  79  75185  99753.031   9606.044
 R   2 G  80  75079  99062.074   9428.958
 R   2 G  81  70572  98828.818   9637.816
 R   2 G  82  75185  98836.258   9582.941
 R   2 G  83  74958  98775.124   9453.874
 R   2 G  84  72162  98863.762   9523.046
 R   2 G  85  74263   98939.73   9574.602
 R   2 G  86  74140   98683.49   9423.708
 R   2 G  87  73348  98910.598    9165.24
 R   2 G  88  73348  99073.306   9532.828
 R   2 G  89  71513  98892.696   9626.718
 R   2 G  90  71513  98892.056   9582.485
 R   2 G  91  71855  98785.527   9673.011
 R   2 G  92  70136  98178.798   9652.116
 R   2 G  93  64963  97849.957   9855.567
 R   2 G  94  67050   98085.93  10391.826
 R   2 G  95  68238  97512.038  10321.606
 R   2 G  96  70476  97493.776  10243.877
 R   2 G  97  69152  97584.165  10175.438
 R   2 G  98  69855  97882.055  10128.495
 R   2 G  99  72606  97556.389   10120.65
 R   2 G 100  68822  97574.681   9946.498
 R   2 G 101  70262  97557.333  10053.823
 R   2 G 102  70262  97419.195   9919.815
 R   2 G 103  61539  97393.369   9991.794
 R   2 G 104  63428  97338.697   9686.404
 R   2 G 105  63428   97662.37   9737.743
 R   2 G 106  63428  97287.686  10122.162
 R   2 G 107  68370  97090.671  10077.335
 R   2 G 108  70844  97011.348   9822.934
 R   2 G 109  66185  97057.498  10003.189
 R   2 G 110  66185   96591.32  10049.053
 R   2 G 111  70412  96495.776   9792.101
 R   2 G 112  71979  96643.146   9770.732
 R   2 G 113  70363  96827.692   9809.908
 R   2 G 114  69580  96895.371   9770.865
 R   2 G 115  69580  96034.791   9403.147
 R   2 G 116  66697   95896.99   9545.011
 R   2 G 117  69064  96063.743   9562.613
 R   2 G 118  69064  96209.578   9696.566
 R   2 G 119  69710  96181.231   9863.393
 R   2 G 120  68843  96296.575   9654.576
 R   2 G 121  67923  96197.302   9763.429
 R   2 G 122  72273  96457.696   9703.411
 R   2 G 123  65838  96064.027   9843.083
 R   2 G 124  70927  96174.303   9714.901
 R   2 G 125  72247  96393.318   9854.439
 R   2 G 126  71819   96496.02  10045.174
 R   2 G 127  70409  96363.137   9648.945
 R   2 G 128  73635  96461.782   9565.343
 R   2 G 129  69174  96770.651   9735.078
 R   2 G 130  72797  96752.794    9470.39
 R   2 G 131  70051   96670.42   9444.445
 R   2 G 132  70051  96633.618    9518.57
 R   2 G 133  73067  96773.756   9472.435
 R   2 G 134  74019  97207.987   9727.125
 R   2 G 135  71238  97229.473   9736.124
 R   2 G 136  70255   97339.82   9708.057
 R   2 G 137  70255  97061.203   9486.722
 R   2 G 138  70792  97589.387   9657.003
 R   2 G 139  70792  97533.358   9640.243
 R   2 G 140  70792  97380.618   9704.013
 R   2 G 141  70840  97414.763   9881.587
 R   2 G 142  71328  97460.179   9773.788
 R   2 G 143  68115  97142.348    9423.19
 R   2 G 144  68115  97649.551   9768.035
 R   2 G 145  68115  97367.626   9852.464
 R   2 G 146  71133   97134.79    9945.74
 R   2 G 147  67557  97054.574   9803.522
 R   2 G 148  70530  97003.586   9768.794
 R   2 G 149  71133  96588.914   9399.332
 R   2 G 150  71030  96859.177   9780.079
 R   2 G 151  72061  96378.867   9498.852
 R   2 G 152  70751  96500.344   9561.601
 R   2 G 153  70751  96599.159   9827.742
 R   2 G 154  70953  96452.047   9728.959
 R   2 G 155  70953  96482.795    9894.57
 R   2 G 156  70262  96665.582  10014.815
 R   2 G 157  71623  96098.967   9807.087
 R   2 G 158  69708  96133.777   9910.575
 R   2 G 159  68645  95819.673   9736.488
 R   2 G 160  71416  95977.041   9543.623
 R   2 G 161  71807  95882.164   9427.832
 R   2 G 162  71807  95879.056   9683.615
 R   2 G 163  72219  95806.664   9519.592
 R   2 G 164  65724  95453.004   9357.555
 R   2 G 165  70692  95773.829   9788.654
 R   2 G 166  70843  95257.534   9675.832
 R   2 G 167  71423  95013.507   9815.649
 R   2 G 168  69512  94741.001   9448.884
 R   2 G 169  69512  95138.889  10189.855
 R   2 G 170  68140  94633.073   9822.596
 R   2 G 171  71112  94313.222   9702.018
 R   2 G 172  68590  94199.425   9696.129
 R   2 G 173  66404  94150.006   9542.102
 R   2 G 174  71254  94317.353   9578.522
 R   2 G 175  69509  94167.005   9402.727
 R   2 G 176  71728  94414.968   9341.639
 R   2 G 177  68749  94352.384   9748.519
 R   2 G 178  71212  94339.817     9610.7
 R   2 G 179  71107  94131.392   9719.506
 R   2 G 180  68860  94154.458   9583.904
 R   2 G 181  68860  93801.095   9370.871
 R   2 G 182  67207  93965.612   9416.901
 R   2 G 183  71374  93689.766   9089.995
 R   2 G 184  69530  93817.505   9265.551
 R   2 G 185  67327  93941.091   9367.802
 R   2 G 186  69516  93678.046   9236.239
 R   2 G 187  68404  94276.387   9566.114
 R   2 G 188  67718   94264.85   9787.349
 R   2 G 189  67768  94372.413   9537.386
 R   2 G 190  67751  94505.545   9532.885
 R   2 G 191  69442  94455.407   9622.855
 R   2 G 192  67789  94334.386   9765.176
 R   2 G 193  66151  94182.891   9628.392
 R   2 G 194  69263  94657.464   9435.014
 R   2 G 195  68818  94832.371    9585.51
 R   2 G 196  68818  94711.406   9436.172
 R   2 G 197  67799  94364.129    9418.84
 R   2 G 198  68434  94165.926   9526.487
 R   2 G 199  68598  93913.574   9174.641
2    103         17         30         18         38         31         44         28          6          7         27         43         15         33         20          3          1         29         34          5         48         32         39         10         26          4          2         42         45         35         24         16         41         22         14         13         11          8         12         46          9         40         47         21         25         23         36         19         37   RawFitness
                61539

 R   3 G   0 117215 157580.201  10966.683
 R   3 G   1 117215 152478.236  10250.524
 R   3 G   2 110353 147979.612  10038.492
 R   3 G   3  98004  144483.64  10169.079
 R   3 G   4 112149  141261.98    9814.21
 R   3 G   5 108943  138573.25   9875.128
 R   3 G   6 104872  136603.11   9885.381
 R   3 G   7  94524 134877.719  10158.283
 R   3 G   8  99270 132740.971  10229.068
 R   3 G   9  99306 131021.299   10429.09
 R   3 G  10  97767 129363.421  10520.452
 R   3 G  11  98418 127389.856  10179.638
 R   3 G  12 100307 125814.569   9908.517
 R   3 G  13  96202 124830.078  10378.269
 R   3 G  14  96391 123303.039   10419.12
 R   3 G  15  93718 122006.521  10560.604
 R   3 G  16  90139 120320.684  10488.194
 R   3 G  17  93565 119190.266  10575.467
 R   3 G  18  91377 118069.736  10259.627
 R   3 G  19  91886   117458.7  10126.622
 R   3 G  20  86840 116666.048  10259.167
 R   3 G  21  88609 116182.349  10555.035
 R   3 G  22  88879 115149.367  10418.482
 R   3 G  23  87790 114234.984  10135.467
 R   3 G  24  87054 114174.991    9933.74
 R   3 G  25  85855 113767.673    9979.49
 R   3 G  26  82901 113971.519  10413.866
 R   3 G  27  81704 113520.708  10634.909
 R   3 G  28  88521 112803.171  10394.606
 R   3 G  29  82305 112125.195  10258.103
 R   3 G  30  81448 111855.871  10565.071
 R   3 G  31  79368 111140.126   10343.53
 R   3 G  32  79368 110051.263  10099.749
 R   3 G  33  79368 110213.237  10029.823
 R   3 G  34  78592 110353.813  10318.336
 R   3 G  35  79598 109870.689  10026.267
 R   3 G  36  78220 109975.969  10312.917
 R   3 G  37  78220 109771.579  10557.874
 R   3 G  38  78220 109101.083  10298.153
 R   3 G  39  78220 108612.008   9972.194
 R   3 G  40  78220  108538.38  10196.396
 R   3 G  41  78220 108208.548   10247.88
 R   3 G  42  78220 107796.313  10482.673
 R   3 G  43  78947 107277.741  10126.722
 R   3 G  44  81778  107149.57  10208.689
 R   3 G  45  76937 106967.295  10213.643
 R   3 G  46  76937 106832.885   10471.82
 R   3 G  47  83131 106264.958   9783.256
 R   3 G  48  81213 106246.425   9829.173
 R   3 G  49  71416 105979.832   9918.067
 R   3 G  50  79536 105698.762   9971.878
 R   3 G  51  79574  105560.56   9602.213
 R   3 G  52  77294   105584.5   9961.155
 R   3 G  53  77100 105326.021   9689.586
 R   3 G  54  77935   105387.0   9979.246
 R   3 G  55  79429 105366.533   9484.979
 R   3 G  56  81247 105353.722   9501.664
 R   3 G  57  76378 104951.454  10088.764
 R   3 G  58  77117 104153.634   9670.391
 R   3 G  59  77877 103456.361   9557.863
 R   3 G  60  77056 103215.274   9872.241
 R   3 G  61  76602 102773.549   9842.836
 R   3 G  62  77592 102219.292    9688.58
 R   3 G  63  69684 101618.802   9941.586
 R   3 G  64  74725  101443.97   9697.232
 R   3 G  65  75136 101032.783   9698.219
 R   3 G  66  66053 100833.038   9823.521
 R   3 G  67  66849 100655.662   9449.014
 R   3 G  68  66849 100764.705   9467.549
 R   3 G  69  67986 100685.897   9402.852
 R   3 G  70  72189 100603.336   9458.813
 R   3 G  71  69113 100464.225   9346.404
 R   3 G  72  69113 100309.307    9593.54
 R   3 G  73  70248   99856.73   9758.677
 R   3 G  74  67780  99479.954   9864.129
 R   3 G  75  71017  99229.716   9782.133
 R   3 G  76  74585  99156.343   9902.754
 R   3 G  77  73511  99291.039   9752.608
 R   3 G  78  75090  99091.699     9764.8
 R   3 G  79  72566  98588.075   9930.063
 R   3 G  80  72082  98784.358   9849.738
 R   3 G  81  69538  98498.174    9801.98
 R   3 G  82  69570  98924.981  10020.863
 R   3 G  83  66666  98657.928   9840.687
 R   3 G  84  66666  98458.458   9898.247
 R   3 G  85  68617  98168.325  10102.667
 R   3 G  86  68617  98550.695  10299.099
 R   3 G  87  72009  98144.039  10161.078
 R   3 G  88  70096  97803.159  10179.065
 R   3 G  89  70278  96781.392  10106.835
 R   3 G  90  68856  95904.337   9680.956
 R   3 G  91  68467   96162.12   9624.327
 R   3 G  92  68467  96464.363   9504.025
 R   3 G  93  72527  96940.692   9610.674
 R   3 G  94  72119  97201.158   9678.792
 R   3 G  95  72119  97371.072   9885.896
 R   3 G  96  73400  97236.082   9700.712
 R   3 G  97  73773  96875.042   9601.756
 R   3 G  98  73203  96780.952   9839.951
 R   3 G  99  68318  97112.605   9876.641
 R   3 G 100  68318   97101.49   9906.145
 R   3 G 101  70894  97398.377    9998.65
 R   3 G 102  69067  97078.675   9801.321
 R   3 G 103  70523  97126.114  10164.694
 R   3 G 104  65628  96915.901   9756.449
 R   3 G 105  69322  97423.134  10075.191
 R   3 G 106  69322  97800.443  10062.066
 R   3 G 107  68821  97470.776   9859.839
 R   3 G 108  66710  97085.079  10282.788
 R   3 G 109  66686  96843.251  10190.015
 R   3 G 110  70009  96631.118  10196.789

Ordinal_Representation/Parameters.java

/******************************************************************************
*  A Teaching GA					  Developed by Hal Stringer & Annie Wu, UCF
*  Version 2, January 18, 2004
*******************************************************************************/

import java.io.*;
import java.util.*;
import java.text.*;

public class Parameters {

	/*******************************************************************************
	 * INSTANCE VARIABLES *
	 *******************************************************************************/

	public static String expID;
	public static String problemType;

	public static String dataInputFileName;

	public static int numRuns;
	public static int generations;
	public static int popSize;

	public static int genCap;
	public static int fitCap;

	public static String minORmax;
	public static int selectType;
	public static int scaleType;

	public static int xoverType;
	public static double xoverRate;
	public static int mutationType;
	public static double mutationRate;
	public static double DMWindowBegin;
	public static double DMWindowEnd;

	public static long seed;
	public static int numGenes;
	public static int geneSize;

	public static double distance[][]; // distance[City_A][City_B] == distance[City_B][City_A]

	public static ArrayList<Integer> canonicTour;

	/*******************************************************************************
	 * CONSTRUCTORS *
	 *******************************************************************************/

	public Parameters(String parmfilename) throws java.io.IOException {

		String readLine;
		BufferedReader parmInput = new BufferedReader(new FileReader(parmfilename));

		expID = parmInput.readLine().substring(30);// Experiment ID
		problemType = parmInput.readLine().substring(30);

		dataInputFileName = parmInput.readLine().substring(30);

		numRuns = Integer.parseInt(parmInput.readLine().substring(30).trim());
		generations = Integer.parseInt(parmInput.readLine().substring(30).trim());// per run
		popSize = Integer.parseInt(parmInput.readLine().substring(30).trim());

		selectType = Integer.parseInt(parmInput.readLine().substring(30).trim());
		// 1 = Proportional Selection
		// 2 = Tournament Selection
		// 3 = Random Selection
		// 4 = Rank Selection

		scaleType = Integer.parseInt(parmInput.readLine().substring(30).trim());// Fitness Scaling Type
		// 0 = Scale for Maximization (no change to raw fitness)
		// 1 = Scale for Minimization (reciprocal of raw fitness)
		// 2 = Rank for Maximization
		// 3 = Rank for Minimization

		xoverType = Integer.parseInt(parmInput.readLine().substring(30).trim());
		// 1 = Single Point Crossover
		// 2 = Two Point Crossover
		// 3 = Uniform Crossover

		xoverRate = Double.parseDouble(parmInput.readLine().substring(30).trim()); // from 0 to 1, Use "0" to turn off
																					// crossover

		mutationType = Integer.parseInt(parmInput.readLine().substring(30).trim());// 1 = Flip Bit

		mutationRate = Double.parseDouble(parmInput.readLine().substring(30).trim());// from 0 to 1, Use "0" to turn off

		DMWindowBegin = Double.parseDouble(parmInput.readLine().substring(30).trim());
		DMWindowEnd = Double.parseDouble(parmInput.readLine().substring(30).trim());

		// mutation

		seed = Long.parseLong(parmInput.readLine().substring(30).trim());

		numGenes = Integer.parseInt(parmInput.readLine().substring(30).trim());// in each chromosome.

		geneSize = Integer.parseInt(parmInput.readLine().substring(30).trim());
		// is the number of bits in each gene. Number of Genes times Size
		// gives the number of bits in each chromosome.

		// Read cities
		System.out.println("\nInput File Name is: " + dataInputFileName + "\n");
		try (BufferedReader br = new BufferedReader(new FileReader(dataInputFileName))) {
			String line;
			String dimensionField = "DIMENSION : ";
			int numCities = 0;
			// Find dimension field
			while ((line = br.readLine()) != null) {
				if (line.startsWith(dimensionField)) {
					numCities = Integer.parseInt(line.substring(dimensionField.length()));
					// Skip next 2 lines:
					br.readLine();
					br.readLine();
					break;
				}
			}
			if (numCities == 0) {
				System.out.println("\nFailed to read cities. Please check name of input file\n");
				return;
			} else {
				System.out.println("\n" + numCities + "cities found.\nPlease wait...");
			}
			distance = new double[numCities][numCities]; // city 0 is a ghost city

			String cityInfo[];
			int i = 0, x = 0, y = 1;
			double citiesCoordinates[][] = new double[numCities][2];

			while (i < numCities) {
				line = br.readLine();
				cityInfo = line.split(" ");
				citiesCoordinates[i][x] = Double.valueOf(cityInfo[1]); // x coordinate
				citiesCoordinates[i][y] = Double.valueOf(cityInfo[2]); // y coordinate
				i++;
			}

			for (int j = 0; j < numCities; j++) {
				for (int k = 0; k < numCities; k++) {
					distance[j][k] = Distance(citiesCoordinates[j][x], citiesCoordinates[j][y], citiesCoordinates[k][x],
							citiesCoordinates[k][y]);
					// System.out.println("Calculating distance between: City "j + " and " + k +
					// "\n");
				}
			}
			System.out.println("\nDistances between all cities calculated successfully \n");

			canonicTour = new ArrayList<Integer>(numGenes);
			for (int j = 0; j < numGenes; j++) {
				canonicTour.add(j);
			}

			// Comment out this line when not debugging
			// System.out.println(Arrays.deepToString(distances));
		}

		parmInput.close();

		if (scaleType == 0 || scaleType == 2)
			minORmax = "max";
		else
			minORmax = "min";

	}

	/*******************************************************************************
	 * MEMBER METHODS *
	 *******************************************************************************/

	/*******************************************************************************
	 * STATIC METHODS *
	 *******************************************************************************/

	public static double Distance(double x1, double y1, double x2, double y2) {
		return Math.sqrt((y2 - y1) * (y2 - y1) + (x2 - x1) * (x2 - x1));
	}

	public static void outputParameters(FileWriter output) throws java.io.IOException {

		output.write("Experiment ID                :  " + expID + "\n");
		output.write("Problem Type                 :  " + problemType + "\n");

		output.write("Data Input File Name         :  " + dataInputFileName + "\n");

		output.write("Number of Runs               :  " + numRuns + "\n");
		output.write("Generations per Run          :  " + generations + "\n");
		output.write("Population Size              :  " + popSize + "\n");

		output.write("Selection Method             :  " + selectType + "\n");
		output.write("Fitness Scaling Type         :  " + scaleType + "\n");
		output.write("Min or Max Fitness           :  " + minORmax + "\n");

		output.write("Crossover Type               :  " + xoverType + "\n");
		output.write("Crossover Rate               :  " + xoverRate + "\n");
		output.write("Mutation Type                :  " + mutationType + "\n");
		output.write("Mutation Rate                :  " + mutationRate + "\n");

		output.write("Random Number Seed           :  " + seed + "\n");
		output.write("Number of Genes/Points       :  " + numGenes + "\n");
		output.write("Size of Genes                :  " + geneSize + "\n");

		output.write("\n\n");

	}
} // End of Parameters.java **************************************************
